### ISSUE TITLE
How to get the heatmap instead of overlay image?

### ISSUE THREAD

**Issue by kilickaya** [marcoancona/DeepExplain#12]
The method works perfectly fine on a new dataset with another model I trained on. However from what I understand .explain() method returns a (H,W,3) sized gradient image where reddish values correspond to positive evidence and blueish values to negative. 

I wonder how can one directly get the heatmap, instead an overlay image of edge map on top of the heatmap?

**Comment by marcoancona** [comment]
Hi. You can look in the [utils](DeepExplain/examples/utils.py) file how to display the heatmap. You can simply use [Matplotlib's imshow](https://matplotlib.org/api/_as_gen/matplotlib.pyplot.imshow.html) for to get the heatmap from the numpy array. Does this answer your question?

**Comment by kilickaya** [comment]
Thanks for the answer! 

I think I should make it more clear. Given an input image  of (H,W, 3), I want to have a  map for each pixel (H,W) where values close to 1 indicates positive relevance. However, the result returned by, let's say **de.explain('saliency', logits, X, xs)** is a 3-channel map, which is good for visualization but not for quantitative analysis: 

Let's say I want to sort pixels in terms of their positive importance value, so that further I want to perturb the image and observe the accuracy drop (if any). In a way, I want to replicate Figure 3 of https://openreview.net/pdf?id=Sy21R9JAW 

How should I proceed for this?

**Comment by marcoancona** [comment]
I see. You get a 3-channel attribution map because your input is also 3 channels. In other words, you get an attribution score for each input pixels *and each channel*. Now you can decide how to get from here to a single channel heatmap. What is done in the related literature (and also in our paper) is to sum up along the channel dimension (average is also possible).

**Comment by kilickaya** [comment]
Summing across channels makes sense to me. Will try to reproduce results for sensitivity-n analysis. Thanks a lot for that!

**Comment by Subh1m** [comment]
Where can I find the sensitivity-n evaluation source code?

**Comment by Subh1m** [comment]
I found the sensitivity-n code in the shapely branch. It seems we are creating a deltas.hdf5 file for the sensitivity-n test. Can you please explain how to understand the results? I extracted the data from the hdf5 file and got an array for each attribution method.
